engine/src/test/java/org/kigalisim/validate/LiveTests.java

```diff
@@ -340,7 +340,174 @@ public void testBasicUnitsConvert() throws IOException {
+    PushDownMachine machine = new SingleThreadPushDownMachine(engine);
+
+    // Execute the default policy
+    ParsedPolicy defaultPolicy = program.getPolicy("default");
+    assertNotNull(defaultPolicy, "Default policy should exist");
+    executePolicy(defaultPolicy, machine);
+
+    // Execute other policies in the scenario
+    for (String policyName : scenario.getPolicies()) {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Move to year 10 and check results (like the JavaScript test)
+    for (int year = 2; year <= 10; year++) {
+      engine.incrementYear();
+    }
+
+    List<EngineResult> results = engine.getResults();
+
+    // Check Sub A in year 10 - should have 0 GHG consumption due to replacement
+    EngineResult resultSubA10 = findResult(results, "Test", "Sub A", 10);
+    assertNotNull(resultSubA10, "Should have result for Test/Sub A in year 10");
+
+    // Check Sub B in year 10 - should have GHG consumption due to replacement
+    EngineResult resultSubB10 = findResult(results, "Test", "Sub B", 10);
+    assertNotNull(resultSubB10, "Should have result for Test/Sub B in year 10");
+
+    // Verify that Sub B has GHG consumption in year 10 due to replacement
+    System.out.println("RECHARGE ONLY - Sub A Year 10 GHG: " + resultSubA10.getGhgConsumption().getValue() + " " + resultSubA10.getGhgConsumption().getUnits());
+    System.out.println("RECHARGE ONLY - Sub B Year 10 GHG: " + resultSubB10.getGhgConsumption().getValue() + " " + resultSubB10.getGhgConsumption().getUnits());
+    assertTrue(resultSubB10.getGhgConsumption().getValue().doubleValue() > 0,
+        "Sub B should have non-zero GHG consumption in year 10 due to replacement (recharge only)");
+  }
+
+  /**
+   * Debug test to compare sales streams in different scenarios.
+   */
+  @Test
+  public void debugSalesStreamComparison() throws IOException {
+    System.out.println("=== Testing sales stream values ===");
+    
+    // Test 1: Simple case (no retire/recharge)
+    String qtaPathSimple = "/tmp/test_sales_simple.qta";
+    ParsedProgram programSimple = KigaliSimFacade.parseAndInterpret(qtaPathSimple);
+    Engine engineSimple = new SingleThreadEngine(1, 10);
+    PushDownMachine machineSimple = new SingleThreadPushDownMachine(engineSimple);
+    
+    ParsedPolicy defaultPolicySimple = programSimple.getPolicy("default");
+    executePolicy(defaultPolicySimple, machineSimple);
+    
+    // Check sales in year 1
+    engineSimple.setApplication("Test");
+    engineSimple.setSubstance("Sub A");
+    EngineNumber salesSimpleY1 = engineSimple.getStream("sales");
+    System.out.println("SIMPLE - Year 1 Sales: " + salesSimpleY1.getValue() + " " + salesSimpleY1.getUnits());
+    
+    // Move to year 5
+    for (int year = 2; year <= 5; year++) {
+      engineSimple.incrementYear();
```

Comment:
This is not enough to actually run the simulation from year to year. We should transition `LiveTests` as a whole to use `KigaliSimFacade`